### ISSUE TITLE
Dependency Updates - Other Batch

### ISSUE THREAD

**Issue by ParaskP7** [wordpress-mobile/WordPress-Android#17559]
Parent #17551

This issue is about updating all `Other` related dependencies for the whole project.

This `Other` batch contains the following `3` dependencies:

- [x] `stethoVersion = '1.6.0'` (see [releases](https://github.com/facebook/stetho/releases)) -> [Comment with Call to Action](https://github.com/wordpress-mobile/WordPress-Android/issues/17559#issuecomment-1330671609) (❓)
- [x] `androidDesugarVersion = '1.1.5'` (see [tags](https://github.com/google/desugar_jdk_libs/tags)) -> [Comment with Call to Action](https://github.com/wordpress-mobile/WordPress-Android/issues/17559#issuecomment-1330674860) (❓) -> [I Stand Corrected](https://github.com/wordpress-mobile/WordPress-Android/issues/17559#issuecomment-1452021987) (❌) -> #18064
- [x] `wordPressLintVersion = '1.1.0'` (see [repo](https://github.com/wordpress-mobile/WordPress-Lint-Android)) -> [Comment Call to Action](https://github.com/wordpress-mobile/WordPress-Android/issues/17559#issuecomment-1330680761) (❓)

**Comment by ParaskP7** [comment]
✅ The latest stable version of `stethoVersion` is `1.6.0` (see [releases](https://github.com/facebook/stetho/releases)). As such, this item is now marked as complete.

❓ However, maybe we should take this opportunity to upgrade from [Stetho](https://facebook.github.io/stetho/) to [Flipper](https://fbflipper.com/). 🤔

FYI: WCAndroid is already using [Flipper](https://fbflipper.com/) instead.

**Comment by ParaskP7** [comment]
✅ The latest stable version of `androidDesugarVersion` is `1.1.5` (see [tags](https://github.com/google/desugar_jdk_libs/tags)). As such, this item is now marked as complete.

❓ I am not quite sure if we still need this dependency, we should double-check that and move forward accordingly.

FYI: WCAndroid is not using such a dependency, or any `coreLibraryDesugaring` related dependency.

**Comment by ParaskP7** [comment]
❓ I am not sure what's the latest stable dependency for `wordPressLintVersion` (see [repo](https://github.com/wordpress-mobile/WordPress-Lint-Android)). Also, I am not quite sure whether we need these 3 custom Lint rules any more (see below). Maybe we don't and thus we can deprecated this repo as well. 🤔 

Custom Lint Rules:
- [WordPressAndroidImportInViewModelDetector](https://github.com/wordpress-mobile/WordPress-Lint-Android/blob/trunk/WordPressLint/src/main/java/org/wordpress/android/lint/WordPressAndroidImportInViewModelDetector.kt)
- [WordPressIssueRegistry](https://github.com/wordpress-mobile/WordPress-Lint-Android/blob/trunk/WordPressLint/src/main/java/org/wordpress/android/lint/WordPressIssueRegistry.java)
- [WordPressRtlCodeDetector](https://github.com/wordpress-mobile/WordPress-Lint-Android/blob/trunk/WordPressLint/src/main/java/org/wordpress/android/lint/WordPressRtlCodeDetector.java)

**Comment by oguzkocer** [comment]
@ParaskP7 `1.1.0` is the latest version we have for `wordPressLintVersion` in our S3 Maven, so I am fairly certain that it's the latest available version. We had issues adding Buildkite publishing to the [repo](https://github.com/wordpress-mobile/WordPress-Lint-Android), so even if we add/change anything, there is no easy way to publish them at the moment.

I am generally in favor of using commonly available lint rules, rather than custom ones because of the burden of maintaining them. Combining that with the fact that we don't have publishing implemented for the repo, I'd be in favor of dropping this dependency unless there is a significant reason to keep it.

**Comment by ParaskP7** [comment]
> @ParaskP7 1.1.0 is the latest version we have for wordPressLintVersion in our S3 Maven, so I am fairly certain that it's the latest available version.

Thanks for the confirmation on that @oguzkocer ! 🙇 

> We had issues adding Buildkite publishing to the [repo](https://github.com/wordpress-mobile/WordPress-Lint-Android), so even if we add/change anything, there is no easy way to publish them at the moment.

Good to know, thanks for noting that! 🌟 

> I am generally in favor of using commonly available lint rules, rather than custom ones because of the burden of maintaining them.

Same here! 💯 

Especially due to the fact that those Lint rules are old and might not even apply anymore to our project(s), we should at least consider checking them out, assert their values, try to replace them with commonly available Lint rules (if any), then remove this dependency (from this and all other client and lib repos) and stop maintaining this custom Lint repo altogether. 🤔 

> Combining that with the fact that we don't have publishing implemented for the repo, I'd be in favor of dropping this dependency unless there is a significant reason to keep it.

I agree! 💯

**Comment by oguzkocer** [comment]
@ParaskP7 Since `wordPressLintVersion = '1.1.0'` is the latest version, I've marked that as completed and will close this issue. We can follow up on removing the dependency as a separate task when we can.

**Comment by ParaskP7** [comment]
> @ParaskP7 Since wordPressLintVersion = '1.1.0' is the latest version, I've marked that as completed and will close this issue. We can follow up on removing the dependency as a separate task when we can.

I agree @oguzkocer , thank you! 🥇

**Comment by ParaskP7** [comment]
> ✅ The latest stable version of androidDesugarVersion is 1.1.5 (see [tags](https://github.com/google/desugar_jdk_libs/tags)). As such, this item is now marked as complete.

I stand corrected on that because there actually are newer versions of this Android Desugar JDK library (see [Docs](https://developer.android.com/studio/write/java8-support#library-desugaring-versions) and [Google's Maven Repository](https://maven.google.com/web/index.html?q=desug#com.android.tools:desugar_jdk_libs).

> ❓ I am not quite sure if we still need this dependency, we should double-check that and move forward accordingly.

I stand corrected on that too, this Android Desugar JDK library is needed and was introduced in [this PR](https://github.com/wordpress-mobile/WordPress-Android/pull/14698) to help with making date related operations during the [Notifications](https://github.com/orgs/wordpress-mobile/projects/95) project, which the `java.time` API would have made quite painful otherwise.

> FYI: WCAndroid is not using such a dependency, or any coreLibraryDesugaring related dependency.

Once more, I stand correct here too since WCAndroid is about to use this Android Desugar JDK library as well (see [PR](https://github.com/woocommerce/woocommerce-android/pull/8437)), for the very same reasons the WPAndroid team did.

-----

However, updating this library is blocked by the AGP `7.3.0` (for `1.2.2`) and AGP `7.4.0` (for `2.0.2`). Thus, this update isn't possible just yet. For more info on that see this WCAndroid related code review comment [here](https://github.com/woocommerce/woocommerce-android/pull/8437#pullrequestreview-1321967813).
